src/app/carbon-estimation-table/carbon-estimation-table.component.ts

```diff
@@ -0,0 +1,310 @@
+import { ChangeDetectorRef, Component, computed, input } from '@angular/core';
+import { CarbonEstimation } from '../types/carbon-estimator';
+import { EmissionsColours, EmissionsLabels } from '../carbon-estimation/carbon-estimation.constants';
+import { CarbonEstimationUtilService } from '../services/carbon-estimation-util.service';
+import { NumberObject } from '../utils/number-object';
+import { NgClass, NgStyle } from '@angular/common';
+
+export type TableItem = TableItemLevel1 | TableItemLevel2;
+
+type TableItemLevel1 = { level: 1; expanded: boolean } & BaseTableItem;
+type TableItemLevel2 = { level: 2; parent: string; svg: string } & BaseTableItem;
+type BaseTableItem = {
+  category: string;
+  emissions: string;
+  colour: ItemColour;
+  display: boolean;
+  positionInSet: number;
+  setSize: number;
+};
+
+type ItemColour = {
+  svg?: string;
+  background: string;
+};
+
+type ArrowDirection = ArrowDirectionHorizontal | ArrowDirectionVertical;
+
+type ArrowDirectionHorizontal = 'left' | 'right';
+type ArrowDirectionVertical = 'up' | 'down';
+
+@Component({
+  selector: 'carbon-estimation-table',
+  standalone: true,
+  imports: [NgStyle, NgClass],
+  templateUrl: './carbon-estimation-table.component.html',
+})
+export class CarbonEstimationTableComponent {
+  public carbonEstimation = input<CarbonEstimation>();
+
+  public tableData = computed(() => this.getTableData(this.carbonEstimation()));
+
+  private expandedState: { [key: string]: boolean } = {};
+
+  constructor(
+    private carbonEstimationUtilService: CarbonEstimationUtilService,
+    private changeDetector: ChangeDetectorRef
+  ) {}
+
+  public toggle(category: string): void {
+    this.tableData().forEach(emission => {
+      if (emission.level === 2 && emission.parent === category) {
+        emission.display = !emission.display;
+        this.expandedState[category] = emission.display;
```

Comment:
Should the `expandedState` be set from the level 1 if statement? Here it's being updated up to 3 times.

Reply:
Will update